src/main/java/com/klodnicki/Bike/v2/configuration/SecurityConfig.java

```diff
@@ -0,0 +1,57 @@
+package com.klodnicki.Bike.v2.configuration;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.Customizer;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.config.annotation.web.configurers.HeadersConfigurer;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.servlet.util.matcher.MvcRequestMatcher;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.web.servlet.handler.HandlerMappingIntrospector;
+
+@EnableWebSecurity
+@Configuration
+public class SecurityConfig {
+
+    @Bean
+    MvcRequestMatcher.Builder mvc(HandlerMappingIntrospector introspector) {
+        return new MvcRequestMatcher.Builder(introspector);
+    }
```

Comment:
sorry it was unused :)

Reply:
sorry! it turns out it is required! it returns MvcRequestMatcher.Builder that can then be used elsewhere in the application to build MvcRequestMatcher instances, which are typically used for matching HTTP requests :)
It is used here:
![image](https://github.com/BartoliniAndBorderCollies/Bike.v2_REST_API/assets/126821059/67435bcd-e8b9-455d-b21f-91c089fec6d3)

I made a test and deleted this part of code (mvc) , the result was terrible! 

so yes, it is needed :)

Reply:
Would be nice to see what was so terrible about it! I could describe issue we're facing here if I could see the error or whatever is happening :) I know that this bean might be needed in your case but not sure if you do know exactly why. If you're keen to get to the bottom of it, could you send some more details?
If not, just let me know about it and resolve the conversation. Thanks!

Reply:
if I delete it then all integrations test are going to fail, because there is a lack of bean:

Error creating bean with name 'securityFilterChain' defined in class path resource [com/klodnicki/Bike/v2/configuration/SecurityConfig.class]: Unsatisfied dependency expressed through method 'securityFilterChain' parameter 1: No qualifying bean of type 'org.springframework.security.web.servlet.util.matcher.MvcRequestMatcher$Builder' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}

![image](https://github.com/BartoliniAndBorderCollies/Bike.v2_REST_API/assets/126821059/e7694e6f-8e94-4c88-91a9-244a1ccbfbf0)

![image](https://github.com/BartoliniAndBorderCollies/Bike.v2_REST_API/assets/126821059/ff27f49f-131d-44ce-8e4c-35bac8031c83)

Reply:
I would appreciate if you just tell me what exactly it does and why is it so crucial:

![image](https://github.com/BartoliniAndBorderCollies/Bike.v2_REST_API/assets/126821059/72934186-d7d4-48dc-81e1-278162fe3fdf)

Reply:
So the problem is with `@Autowired` on some of your fields in the integration tests - that's why it is crucial for your project.

And if this is required by IT only, then you could put this into a class marked with `@TestConfiguration`.

Reply:
but as far as I can see mvc is used also in SecurityConfig class, here:

![image](https://github.com/BartoliniAndBorderCollies/Bike.v2_REST_API/assets/126821059/36fc77d7-3aa4-48e2-abd0-c98236eee1ce)

When I remove it and put mvc block of code to separate config class just for integration tests it doesn't run, none of the test. The message is the same as I remove mvc block of code at all.

![image](https://github.com/BartoliniAndBorderCollies/Bike.v2_REST_API/assets/126821059/582f41af-1625-4289-a5c4-2c8052c44d9e)


![image](https://github.com/BartoliniAndBorderCollies/Bike.v2_REST_API/assets/126821059/5c68ffbf-b768-4c35-9ce4-4c94458352c1)

Reply:
thanks for this article: https://www.baeldung.com/spring-boot-testing#test-configuration-withtestconfiguration

Now I know how to connect congiguration class for tests with tests itself!

And now it works as it should I believe!